internal/controller/restfuncs.go

```diff
@@ -100,6 +102,9 @@ func commandFunc(w http.ResponseWriter, req *http.Request) {
 	if appErr != nil {
 		http.Error(w, fmt.Sprintf("%s %s", appErr.Message(), req.URL.Path), appErr.Code())
 	} else if event != nil {
+		// push to Core Data
+		go common.SendEvent(event)
```

Comment:
Why did this get moved from command.go? Where we only sending events to Core Data for reads?

Reply:
move this line for the AutoEvent OnChange mechanism, or the event will be sent in command handler anyway

Reply:
This is my biggest concern.  This actually now means we're encoding event into JSON three times, unless I missed something...

If someone calls the command REST endpoint, then we encode the JSON once when `execReadCommand` creates the event. Then, we encode it in `commandFunc` with your new change which calls `common.SendEvent()`, and then again two lines later with: `json.NewEncoder(w).Encode(event)`

See issue #233

Reply:
actually, it won't be three times, and it is a separated issue

Reply:
How many encodings happen is evt != nil in the following code fragment:

```
} else if event != nil {
		// push to Core Data
		go common.SendEvent(event)

		w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
		json.NewEncoder(w).Encode(event)
}
```